src/gluonts/dataset/jsonl.py

```diff
@@ -57,55 +54,22 @@ class JsonLinesFile:
     path
         Path of the file to load data from. This should be a valid
         JSON Lines file.
-    replica_info
-        What worker this dataset is handled by. Default: WorkerInfo()
     """
 
-    def __init__(
-        self, path, replica_info=ReplicaInfo(), burnt_in=False
-    ) -> None:
+    def __init__(self, path) -> None:
         self.path = path
-        self.replica_info = replica_info
-        self._len = None  # cache the calculated length
-        # indicates in the case of cyclic data sets (end_index is None) that the burn in has
-        # been done once: (it is reset whenever the ReplicaInfo() is set)
-        self._burnt_in = burnt_in
+        self._len = None
         # TODO: implement caching here
 
     def __iter__(self):
         with open(self.path) as jsonl_file:
             for line_number, raw in enumerate(jsonl_file):
-                # TODO: I think this iteration logic, as well as total_dataset_len, start_index
-                #  and end_index should be properties of the dataset. Total_dataset_len should be
-                #  metadata. What is done to return an entry should be the only dataset type specific thing.
-
-                # skip until start_index on first pass, aka do burn_in
-                # in case of cyclic data sets always do burn in, in case of non cyclic ones, only once
-                if line_number < self.replica_info.start_index and (
-                    self.replica_info.end_index is not None
-                    or not self._burnt_in
+                # The dataset is equally distributed among the workers
+                if not (
+                    line_number % MPWorkerInfo.num_workers
```

Comment:
Would be more efficient to do this inside the jsonl_file, since we would only read the line but not process it furtherl. But let's change this later, I think we wanted to cleanup the datasets anyway.

Reply:
What do you mean, I though this just reads the line files put doesn't process them further, since it just opens the file like `with open(self.path) as jsonl_file:`?

Reply:
yeah, the only thing done is `json.loads`, which shouldn't be too bad.